### PR TITLE
[FEATURE] Add default TCA columns from ctrl

### PR DESCRIPTION
**Pull request by linawolf**
Releases: main
Resolves: https://github.com/TYPO3-Documentation/Changelog-To-Doc/issues/967

### CODE REVIEW

File: Documentation/BestPractises/CommonFields.rst [hunk 1]
```diff
@@ -16,11 +18,14 @@ If the table has a TCA definition, TYPO3 will automatically create the following
     and as a reference in relationships between records.
 
 :sql:`pid`
-    The `uid` property of the parent page. The record is situated on this page. This value is 0 if the record is not connected to any page.
+    The `uid` property of the parent page. The record is situated on this page.
```
Comment (froemken) [inline Documentation/BestPractises/CommonFields.rst:21]: I know, it was "property" before, but I would prefer to use "property" in extbase context only. For TCA I prefer "column" (of the table) or "field" in BE form.

```suggestion
    The `uid` field of the parent page. The record is situated on this page.
```

File: Documentation/BestPractises/CommonFields.rst [hunk 2]
```diff
@@ -16,11 +18,14 @@ If the table has a TCA definition, TYPO3 will automatically create the following
...
+    If this value is 0 if the record is not connected to any page.
```
Comment (froemken) [inline Documentation/BestPractises/CommonFields.rst:22]:
```suggestion
    If this value is 0 the record is not connected to any page.
```

File: Documentation/Ctrl/AutoCreatedColumns.rst [hunk 3]
```diff
@@ -0,0 +1,79 @@
+.. include:: /Includes.rst.txt
+.. _ctrl-auto-created-columns:
```
Comment (froemken) [inline Documentation/Ctrl/AutoCreatedColumns.rst:2]:
```suggestion
..  include:: /Includes.rst.txt
..  _ctrl-auto-created-columns:
```

File: Documentation/Ctrl/AutoCreatedColumns.rst [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+#.  Apply TCA preparations
+
+As a result of this strategy, :php:`columns` fields are *not* auto-created, when
+a :php:`ctrl` capability is added in a :file:`Configuration/TCA/Overrides`
+file, and *not* in a :file:`Configuration/TCA` "base" file. In general, such
+capabilities should be set in base files only: Adding them at a later point - for
+example in a different extension - is brittle and there is a risk the main
+extension can not deal with such an added capability properly.
+
+.. _ctrl-auto-created-columns-override:
```
Comment (froemken) [inline Documentation/Ctrl/AutoCreatedColumns.rst:49]:
```suggestion
..  _ctrl-auto-created-columns-override:
```

File: Documentation/Ctrl/AutoCreatedColumns.rst [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+There are two options to have own definitions: When a column is already defined
+in a "base" TCA file (:file:`Configuration/TCA`), the Core will not override it.
+Alternatively, a developer can decide to let the Core auto-create a column, to
+then override single properties in :file:`Configuration/TCA/Overrides` files.
+
+As example, "base" :php:`pages` file defines this (step 1 above):
+
+..  literalinclude:: _AutoCreatedColumns/_pages.php
+    :caption:
```
Comment (froemken) [inline Documentation/Ctrl/AutoCreatedColumns.rst:67]: Remove or add a caption

Comment (linawolf) [inline Documentation/Ctrl/AutoCreatedColumns.rst:67]:
```suggestion
    :caption: EXT:core/Configuration/TCA/pages.php (Excerpt)
```

File: Documentation/Ctrl/_AutoCreatedColumns/_OverrrideHiddenDefault.php [hunk 6]
```diff
@@ -0,0 +1,4 @@
+<?php
```
Comment (froemken) [inline Documentation/Ctrl/_AutoCreatedColumns/_OverrrideHiddenDefault.php:1]: This is the file with three `r`

**Review by froemken — CHANGES_REQUESTED** [review]

File: Documentation/Ctrl/_AutoCreatedColumns/_OverrrideHiddenDefault.php (hunk 6 above)
Comment (linawolf) [reply Documentation/Ctrl/_AutoCreatedColumns/_OverrrideHiddenDefault.php:1]: I dont get what you mean?

File: Documentation/Ctrl/AutoCreatedColumns.rst [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+The Core thus creates this :php:`columns` definition (step 2 above):
+
+..  literalinclude:: _AutoCreatedColumns/_OverrrideHiddenDefault.php
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/Ctrl/AutoCreatedColumns.rst:71]: ??

[the remark this answers is not in the document]
Comment (froemken) [reply Documentation/Ctrl/AutoCreatedColumns.rst]: _OverrrrrrrrrrrideHiddenDefault.php ;-) Solved

File: Documentation/Ctrl/Properties/DescriptionColumn.rst [hunk 8]
```diff
@@ -5,24 +5,30 @@
...
+    displayed in the  :guilabel:`Web > Page` and  :guilabel:`Web > List` module, and shown above the field list if
+    editing a record. It is meant as a note field to give editors important
+    additional information on single records. The TYPO3 Core sets this
+    property for a series of main tables like `be_users`, `be_groups` and `tt_content`.
 
+    ..  include:: _AutoCreateWarning.rst.txt
 
-Examples
-========
+.. _ctrl-reference-descriptionColumn-example:
```
Comment (froemken) [inline Documentation/Ctrl/Properties/DescriptionColumn.rst:27]:
```suggestion
..  _ctrl-reference-descriptionColumn-example:
```

File: Documentation/Ctrl/Properties/Enablecolumns.rst [hunk 9]
```diff
@@ -66,6 +98,8 @@ Enablecolumns / enablefields usage
 Most ways of retrieving records in the frontend automatically respect the
 :php:`ctrl->enablecolumns` settings:
 
+.. _enablefields_usage-typoscript:
```
Comment (froemken) [inline Documentation/Ctrl/Properties/Enablecolumns.rst:101]:
```suggestion
..  _enablefields_usage-typoscript:
```

File: Documentation/Ctrl/Properties/Enablecolumns.rst [hunk 10]
```diff
@@ -74,6 +108,8 @@ Records retrieved in TypoScript via the objects
 automatically respect the settings in section
 :ref:`ctrl->enablecolumns <ctrl-reference-enablecolumns>`.
 
+.. _enablefields_usage-extbase:
```
Comment (froemken) [inline Documentation/Ctrl/Properties/Enablecolumns.rst:111]:
```suggestion
..  _enablefields_usage-extbase:
```

File: Documentation/Ctrl/Properties/Enablecolumns.rst [hunk 11]
```diff
@@ -82,6 +118,8 @@ however this behaviour can be disabled by setting
 :php:`$querySettings->setIgnoreEnableFields(true)` in the
 :ref:`repository <t3coreapi:extbase-repository>`.
 
+.. _enablefields_usage-queries:
```
Comment (froemken) [inline Documentation/Ctrl/Properties/Enablecolumns.rst:121]:
```suggestion
..  _enablefields_usage-queries:
```

**Review by froemken — APPROVED** [review]

**Pull request merged**